### PR TITLE
[SNOW-116] New template repository for Streamlit apps

### PR DESCRIPTION
**Pull request by jaymedina**
I'm migrating the [Streamlit app module ](https://github.com/Sage-Bionetworks/snowflake/tree/dev/streamlit_template)to be its own template repository that new repositories can be based off of. Moving it out of the `snowflake` repository will minimize risk of other areas in the repository accidentally getting modified, and it will allow non-DPE users to create and contribute to their own app repositories with minimal overhead, allowing the focus of the template app to be solely a template app.

## Changelog
- [x] Migrate all components from `snowflake` to this repository branch
- [x] Update the `README` with new instructions for how users can make their apps

### CODE REVIEW

**Comment by BryanFauble** [comment]
Should we copy over `.vscode` as well?

https://github.com/Sage-Bionetworks/snowflake/tree/dev/.vscode

**Comment by BryanFauble** [comment]
You'll also want to copy over:

https://github.com/Sage-Bionetworks/snowflake/blob/dev/.github/CODEOWNERS

And double check the permissions. Do we have a standard around granting X/Y groups admin rights on these? Like IT, DPE, individuals ect... @thomasyu888 ?

**Review by BryanFauble — CHANGES_REQUESTED** [review]
A few small things to double check before we merge this in

**Comment by jaymedina** [comment]
Updated, thanks @BryanFauble - and as for permissions, right now the DPE team has maintainer permissions. My inclination is to give IT and Tom admin permissions.

File: .vscode/launch.json
Comment (BryanFauble) [inline .vscode/launch.json:11]: directory paths will need to be updated here, L13, L23, L25 to match the new directory structure. `cwd` can probably be dropped since everything can just run at the root of the project now instead of it being in a sub directory.

**Review by BryanFauble — DISMISSED** [review]
Just 1 last comment, otherwise looks good to me!

Makes sense to add tom/IT for permissions here.

**Comment by jaymedina** [comment]
Thanks @BryanFauble ! My ruleset is a little strict. Could you take a look at my last commit and give me another review?

**Review by BryanFauble — APPROVED** [review]
LGTM!

**Pull request merged**
